app/src/auth/components/SessionProvider/index.jsx

```diff
@@ -16,23 +18,40 @@ type Props = {
 const storage = isLocalStorageAvailable() ? window.localStorage : null
 
 let cachedToken // fallback if no webstorage
+let cachedDate
 
 function getStoredToken(): ?string {
-    if (!storage) { return cachedToken || null }
-    return storage.getItem(SESSION_TOKEN_KEY) || null
+    let date
+    let token
+    if (!storage) {
+        token = cachedToken || null
+        date = cachedDate || null
+    } else {
+        token = storage.getItem(SESSION_TOKEN_KEY) || null
+        date = storage.getItem(SESSION_EXPIRES_AT) || null
```

Comment:
Perhaps if we store the time that the session *started*, rather than the time it ends, then we could adjust the `EXPIRES_AT_VALID_HOURS` in the future and that change would work immediately, even for tokens stored with the previous `EXPIRES_AT_VALID_HOURS`?

Reply:
That's a good point, changed to set login time in https://github.com/streamr-dev/streamr-platform/pull/942/commits/f005eb0ca7e7e7ba9f149f8bc5a1bd7b00047381